docs/upgrade.md

```diff
@@ -0,0 +1,245 @@
+
+## Automatic upgrade
+
+To upgrade your Storybook:
+
+```sh
+npx storybook@next upgrade --prerelease
+```
+
+This will:
+
+1. Upgrade your Storybook dependencies to the latest prerelease version
+2. Run a collection of _automigrations_, which will:
+   - Check for common upgrade tasks
+   - Explain the necessary changes with links to more information
+   - Ask for approval, then perform the task on your behalf
+
+To add Storybook to a project that isn’t currently using Storybook:
+
+```sh
+npx storybook@next init
+```
+
+This will:
+
+1. Figure out which renderer (React, Vue, Angular, Web Components), builder (Webpack, Vite), or meta-framework (Next.js, SvelteKit) you’re using
+2. Install Storybook 7 and auto-configure it to mirror project settings
+
+## Manual migrations
+
+In addition to the automated upgrades above, there are manual migrations that might be required to get Storybook 7 working in your project. We’ve tried to minimize this list to make it easier to upgrade. These include:
+
+- Upgrade MDX1 to MDX2
+- `storiesOf` support discontinued by default
+
+### Upgrade MDX1 to MDX2
+
+Storybook 7 uses MDX2 by default for rendering [docs](./writing-docs/introduction.md). The upgrade from MDX1 to MDX2 is not fully automated, due to the large number of changes between versions. Fortunately, we have some tips to help make it a lot easier.
+
+#### Automatically detect MDX2 errors with a CLI tool
+
+If your project contains MDX files, run the following command before starting up Storybook:
+
+```sh
+npx @hipster/mdx2-issue-checker
+```
+
+This will go through every MDX file in the current working directory, and show you which files have errors:
+
+![Terminal output of mdx2-issue-checker, showing errors found](./assets/mdx2-issue-checker-errors.png)
+
+#### Fix MDX2 errors iteratively
+
+The CLI only shows the first error per file, so you might need to run the checker iteratively. One way to streamline this process if you use VS Code is to:
+
+1. Install the [MDX VS Code extension](https://marketplace.visualstudio.com/items?itemName=unifiedjs.vscode-mdx)
+2. Enable experimental IntelliSense support for MDX files in your user settings: `"mdx.experimentalLanguageServer": true`
+
+This shows the errors visually in your editor, which speeds things up a lot. Here's what it looks like to fix multiple errors in a file using the extension:
+
+![MDX errors showing in VS Code](./assets/mdx-vs-code-extension-errors.gif)
+
+#### MDX1 as fallback
+
+If, for some reason, you are unable to get MDX2 working, we’ve implemented legacy MDX1 support as a last resort. MDX1 is deprecated and opt-in, and we recommend against it unless you really need it.
+
+To use MDX1:
+
+1. Install `@storybook/mdx1-csf@next` as a dev dependency
+2. Add the `legacyMdx1` feature flag to your `.storybook/main.js`:
+
+```js
+export default {
+  features: {
+    legacyMdx1: true,
+  },
+};
```

Comment:
Can you update this and the other code examples to use the snippet files we currently have? I believe this and the other examples should have a corresponding file. And what's more, it would prevent readers from getting a JS version of the example when they've chosen the TS version in the language selector

Reply:
Thanks for the reminder!